src/app/routes/verifiable-presentation/verifiable-presentation/verifiable-presentation.component.ts

```diff
@@ -26,34 +35,35 @@ export class VerifiablePresentationComponent implements OnInit {
     private store: Store,
     public dialog: MatDialog,
     private iamService: IamService
-  ) {}
+  ) { }
   isAutolistLoading = {
     requests: [],
     value: false,
   };
   ngOnInit(): void {
-    this._initSearch();
+    this._initPresentationFetch();
   }
-  private _initSearch() {
-    this.route.queryParams.subscribe(async (params: any) => {
-      //this.loadingService.show()
+  _initPresentationFetch() {
+    this.route.queryParams.subscribe(async (params: {_oob?: string}) => {
       await this.initLoginUser();
       if (params._oob) {
         const paramsDecoded = atob(params._oob);
         const parsedToObj = JSON.parse(paramsDecoded);
+        console.log(JSON.stringify(parsedToObj), "THE PARAMS!!!")
         const { url } = parsedToObj;
-        console.log(url);
         /*
-                TO DO: Call ICL when persistence of pres def is implemented:
-                    try {
-                    await this.iamService.verifiableCredentialsService.initiateExchange({type: 'https://energyweb.org/out-of-band-invitation/vc-api-exchange', url: url})
-                    this.loadingService.hide();
-                } catch (e) {
-                    console.log(e)
-                } finally {
-                    this.loadingService.hide();
-                }
-                 */
+         TO DO:
+         - Error handling - no url parsed
+         - UI Handling - no results found from initiate exchange
+        */
+        console.log(url);
+        const result = await this.iamService.verifiableCredentialsService.initiateExchange({ type: 'https://energyweb.org/out-of-band-invitation/vc-api-exchange', url });
```

Comment:
move getting and formatting all this data to separate service. It will be easier to test because you do not need to mock queryParams. This method can return object like: 
`{ challenge: ..., interact: ..., presDef: ..., data: ...}`.

Reply:
Oh I see you mean like asset.service.ts as an example. got it!